qiskit_nature/second_q/operators/tensor.py

```diff
@@ -192,6 +201,116 @@ def ndim(self) -> int:
 
+    @property
+    def label_template(self) -> str:
+        """The template string used during the translation implemented in
+        :meth:`.SparseLabelOp.from_polynomial_tensor`.
+
+        By default, any ``Tensor`` instance will return the same string here. This returned template
+        depends on the dimension of the wrapped matrix. It will repeat ``{}_{{}}`` for every
+        dimension. This is explained best with an example:
+
+        .. code-block:: python
+
+            print(Tensor(np.eye(2)).label_template)
+            # "{}_{{}} {}_{{}}"
+
+            print(Tensor(np.ones((2, 2, 2, 2)).label_template)
+            # "{}_{{}} {}_{{}} {}_{{}} {}_{{}}"
+
+        The format of this template only makes sense in the context of what is done inside of
+        :meth:`.SparseLabelOp.from_polynomial_tensor`. There, the template is processed in two
+        steps:
+
+        1. First, the template is formatted using the key under which the ``Tensor`` object was
+           stored inside of the :class:`.PolynomialTensor` object. For example:
+
+           .. code-block:: python
+
+               poly = PolynomialTensor(
+                  {
+                      "+-": Tensor(np.eye(2)),
+                      "++--": Tensor(np.ones((2, 2, 2, 2))),
+                  }
+              )
+
+               # the label_template will get expanded like so:
+               for key, tensor in poly.items():
+                  sparse_label_template = tensor.label_template.format(*key)
+                  print(key, "->", sparse_label_template)
+
+              # "+-" -> "+_{} -_{}"
+              # "++--" -> "+_{} +_{} -_{} -_{}"
+
+        2. Next, these templates are used to build the actual labels of the :class:`.SparseLabelOp`
+           being constructed. For that, the indices encountered during :meth:`coord_iter` are used
+           for example like so:
+
+           .. code-block:: python
+
+              sparse_label_template = "+_{} -_{}"
+              for value, index in Tensor(np.eye(2)).coord_iter():
+                  sparse_label = sparse_label_template.format(*index)
+                  print(sparse_label, value)
+
+              # "+_0 -_0", 1
+              # "+_0 -_1", 0
+              # "+_1 -_1", 1
+              # "+_1 -_0", 0
+
+        Given that you now understand how the ``label_template`` attribute is being used, this
+        allows you to modify how the ``Tensor`` objects stored inside a :class:`.PolynomialTensor`
+        are processed when they are being translated into a :class:`.SparseLabelOp`.
+
+        .. note::
+
+           The string formatting in Python is a very powerful tool `[1]`_. Note, that in the use
+           case here, we only ever supply positional arguments in the ``.format(...)`` calls which
+           means that you cannot use names to identify replacement fields in your label templates.
+           However, you can modify their replacement order using numeric values (like shown below).
```

Comment:
That is a good point. I will try to come up with a phrasing which hints at this but does not restrict it to this particular example (because there are also more general use cases one can think of)